k9mail/src/main/java/com/fsck/k9/message/MessageBuilder.java

```diff
@@ -235,30 +228,32 @@ private void addAttachmentsToMessage(final MimeMultipart mp) throws MessagingExc
              * MimeHeader.writeTo().
              */
             bp.addHeader(MimeHeader.HEADER_CONTENT_TYPE, String.format("%s;\r\n name=\"%s\"",
-                    contentType,
+                    attachment.contentType,
                     EncoderUtil.encodeIfNecessary(attachment.name,
                             EncoderUtil.Usage.WORD_ENTITY, 7)));
 
-            bp.setEncoding(MimeUtility.getEncodingforType(contentType));
-
-            /*
-             * TODO: Oh the joys of MIME...
-             *
-             * From RFC 2183 (The Content-Disposition Header Field):
-             * "Parameter values longer than 78 characters, or which
-             *  contain non-ASCII characters, MUST be encoded as specified
-             *  in [RFC 2184]."
-             *
-             * Example:
-             *
-             * Content-Type: application/x-stuff
-             *  title*1*=us-ascii'en'This%20is%20even%20more%20
-             *  title*2*=%2A%2A%2Afun%2A%2A%2A%20
-             *  title*3="isn't it!"
-             */
-            bp.addHeader(MimeHeader.HEADER_CONTENT_DISPOSITION, String.format(Locale.US,
-                    "attachment;\r\n filename=\"%s\";\r\n size=%d",
-                    attachment.name, attachment.size));
+            if (!MimeUtil.isMessage(attachment.contentType)) {
+                bp.setEncoding(MimeUtility.getEncodingforType(attachment.contentType));
```

Comment:
This is related to #1034. Although reading how qmail and exim handle this issue ([source](https://en.wikipedia.org/wiki/8-bit_clean)) we might want to rethink this.

We need to come up with a concept on 8bit messages sooner or later, let's continue the discussion over there